src/core/lib/slice/slice.cc

```diff
@@ -399,7 +399,8 @@ grpc_slice grpc_slice_split_tail(grpc_slice* source, size_t split) {
   return grpc_slice_split_tail_maybe_ref(source, split, GRPC_SLICE_REF_BOTH);
 }
 
-grpc_slice grpc_slice_split_head(grpc_slice* source, size_t split) {
+template <bool do_ref = true>
```

Comment:
Since this is internal only, can you avoid a default parameter? It obscures intent IMO. It looks like the default is never used anyway.